[19.0][MIG] mail_activity_board: Migration to 19.0

bizzappdev opened the pull request.

dajuayen committed:
[ADD][mail_activity_board] Add new module that insert activities board in boards. (#283)

* [ADD] Module that insert activities board.

* [FIX] Author error in __manifest__ file and style changes.

* [FIX] Fix replace in view, rename files and style changes.

* [FIX] Enumerated list ends without a blank line; unexpected unindent.

* [FIX] Name fail.

* [FIX] Bug in view.

* [FIX]  Add button Activities in mail.thread and readme folder. Others improvements in style of code.

* [FIX] Type 'tree' not found in registry: problem solved.

* [FIX] Dependence change: 'mail' for 'calendar'.

* [FIX] Eliminated unnecessary imports.

* [FIX] Bugs about js and if/else.

* [FIX] Improvements following guide lines and eliminating unnecessary attributes in views.

* [ADD] Added counter in the 'Activities List' button.

* [FIX] Bugs in javascript with 'Activities' button.

* [ADD] Tests folder.

* [FIX] Deleted references to modules not installed.

* [FIX] Formatting javascript.

* [FIX] Bug: added a soft line before a class.

* [FIX] Bug: https://github.com/OCA/social/pull/283#discussion_r204302325

* [FIX] Escaping 'lt' in xml file. Bug: https://github.com/OCA/social/pull/283#discussion_r204302325

* [FIX] The meeting attendees are shown in kanban mode on the meeting board.

* [FIX] Hide in form view of the activity board the assistant field if the activity is not a meeting type or if there are no assistants.

* [FIX] Change to default kanban view for partners.

MiquelRForgeFlow committed:
[IMP] mail_activity_board: filter activities by user read permissions

MiquelRForgeFlow committed:
[MIG] mail_activity_board: Migration to 12.0

pegonzalezspesol committed:
[MIG] 13.0 mail_activity_board

lmignon committed:
[IMP] mail_activity_board: Direct access to related model instance
Add a direct access from activity calendar and form views to the linked model instance.

Bedore this change, it was not possible to access to the model instance linked to an activity from the calendar view nor from the view opened by clicling on the activity into the calendar

[IMP] mail_activiy_board: Restrict related_model_instance selection to models supporting mail acitvities

Co-Authored-By: Miquel Raïch <miquel.raich@eficent.com>

Commit:
Added translation using Weblate (French)

eduardgm committed:
Added translation using Weblate (Spanish)

MiquelRForgeFlow committed:
[FIX] solved: unlink issue due to bad migration

atchuthan committed:
[IMP] : black, isort, prettier

atchuthan committed:
[MIG] mail_activity_board: Migration to 14.0

[FIX] add title to <i> tag and removed console.log in js

Co-authored-by: Miquel Raïch <miquel.raich@forgeflow.com>

Co-authored-by: Miquel Raïch <miquel.raich@forgeflow.com>

JoanMForgeFlow committed:
[15.0][MIG] mail_activity_board: Migration to 15.0

tfossoul committed:
Added translation using Weblate (French)

jabelchi committed:
Added translation using Weblate (Catalan)

AaronHForgeFlow committed:
[FIX] mail_activity_board: read access to ir.model for non admin
users when accessing records that inherit mail.activity mixing

augusto-weiss committed:
[MIG] mail_activity_board: Migration to 16.0

victoralmau committed:
[IMP] mail_activity_board: Add group so that the activity dashboard is only visible if the user has the permission.

TT43037

houzefa-abba committed:
Translated using Weblate (French)

Currently translated at 95.4% (21 of 22 strings)

Translation: social-16.0/social-16.0-mail_activity_board
Translate-URL: https://translation.odoo-community.org/projects/social-16-0/social-16-0-mail_activity_board/fr/

ypapouin committed:
[FIX] mail_activity_board: Use valid field for user avatar

`image_small` field has been renamed to `image_small` since the use of `image.mixin`

mymage committed:
Added translation using Weblate (Italian)

lef-adhoc committed:
[IMP] mail_activity_board: pre-commit auto fixes

lef-adhoc committed:
[MIG] mail_activity_board: Migration to 17.0

Commit:
[UPD] copier version 1.20

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (22 of 22 strings)

Translation: social-17.0/social-17.0-mail_activity_board
Translate-URL: https://translation.odoo-community.org/projects/social-17-0/social-17-0-mail_activity_board/it/

pedrocs-exo committed:
Added translation using Weblate (Portuguese)

astirpe committed:
[17.0][FIX] mail_activity_board: hide button if has no activities

Commit:
[IMP] mail_activity_board: pre-commit stuff

TelmoSuarez committed:
[FIX] mail_activity_board: add no_open to avoid accessing the record and using the open_origin function instead.

kafai-lam committed:
[IMP] mail_activity_board: update user_id widget

StefanRijnhart committed:
[FIX] mail_activity_board: calendar view cannot render reference fields anymore

The regular relational model fetches records using web_search_read (https://github.com/odoo/odoo/blob/7c640426903bc6fa4c972406df1f0905557cf7e5/addons/web/static/src/model/relational_model/relational_model.js#L582). This method formats reference fields as

{ id: { id: 27, model: "res.partner"}, display_name: "Azure Interior, Brandon Freeman"}

(https://github.com/odoo/odoo/blob/7c640426903bc6fa4c972406df1f0905557cf7e5/addons/web/models/models.py#L220-L224)

This structure is used when rendering the relational model widget (https://github.com/odoo/odoo/blob/7c640426903bc6fa4c972406df1f0905557cf7e5/addons/web/static/src/model/relational_model/utils.js#L467-L475).

However, the calendar web model fetches records using search_read (https://github.com/odoo/odoo/blob/7c640426903bc6fa4c972406df1f0905557cf7e5/addons/web/static/src/views/calendar/calendar_model.js#L489). This returns values for reference fields in the conventional format "res.partner,27", but they are still rendered by the same widget when the calendar popup view is opened, resulting in an error 'TypeError: value.id is undefined'.

Commit:
[MIG] mail_activity_board: Migration to 18.0

Co-authored-by: Stefan Rijnhart <stefan@opener.amsterdam>

bizzappdev committed:
[MIG] mail_activity_board: Migration to 19.0

CRogos committed:
[IMP] mail_activity_board: remove spreadsheet_dashboard dependency and apply dashboard changes to existing views

CRogos reviewed (commented):
Could you have a look on these improvements. This module is a little messy :( 
https://github.com/OCA/mail/pull/120

atchuthan reviewed (approved):
Functional review :+1:

CRogos reviewed (changes requested):
We should wait for / add this PR: https://github.com/OCA/mail/pull/120

RicardCForgeFlow committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (22 of 22 strings)

Translation: mail-18.0/mail-18.0-mail_activity_board
Translate-URL: https://translation.odoo-community.org/projects/mail-18-0/mail-18-0-mail_activity_board/ca/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: mail-18.0/mail-18.0-mail_activity_board
Translate-URL: https://translation.odoo-community.org/projects/mail-18-0/mail-18-0-mail_activity_board/

CRogos reviewed (changes requested):
https://github.com/OCA/mail/pull/120 has been merged. Please add to migration.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (10 of 10 strings)

Translation: mail-18.0/mail-18.0-mail_activity_board
Translate-URL: https://translation.odoo-community.org/projects/mail-18-0/mail-18-0-mail_activity_board/it/

victoralmau committed:
[FIX] mail_activity_board: Set view_mode+act_window_id

Similar to https://github.com/OCA/mail/commit/ff68038a8340a8c4369c0edbf519344de1130355

victoralmau commented:
Please, cherry-pick https://github.com/OCA/mail/pull/138 to commit history (before migration commit).

bizzappdev committed:
[IMP] mail_activity_board: pre-commit auto fixes

isturiz commented:
Hi @bizzappdev, ready to review?

MohamedOsman7 commented:
@bizzappdev Could you mark this PR as ready for review?

CRogos commented:
@pedrobaeza can this be merged even though the PR is draft? I think this is ready.

pedrobaeza commented:
/ocabot migration mail_activity_board
/ocabot merge nobump
